processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperLogLogCollector.java

```diff
@@ -92,8 +92,23 @@ public static HyperLogLogCollector makeLatestCollector()
 
   public static HyperLogLogCollector makeCollector(ByteBuffer buffer)
   {
+    return makeCollector(buffer, false);
+  }
+
+  /**
+   * Create a wrapper object around an HLL sketch contained within a buffer.
+   *
+   * @param buffer      buffer containing HLL sketch
+   * @param bufferOwned if true, we are permitted to change the position and limit of the buffer
```

Comment:
I propose changing the contract here such that ONLY buffers who can have their limit and position changed can be passed.